### DISCUSSION TITLE
Mouth movement when dialogue is typing.

### DISCUSSION THREAD

**Discussion by TunaNoot** [dialogic-godot/dialogic#2109] [Q&A] [answered]
Hey, I was wondering if there was a way to display an animated sprite of a character's mouth moving for when dialogue is actively being typed in a box, and having it switch to a different "silent" character sprite when the dialogue box is finished displaying all of the text. Something like Phoenix Wright or 999 visually. I'm sorry if I'm not very eloquent, but I really feel this little bit of animation will go a long way in bringing my characters to life.

**Comment by CakeVR** [comment, marked as answer]
Hello, yes, this can be done via custom portrait scenes: https://docs.dialogic.pro/custom-portraits.html

I use [text_finished](https://docs.dialogic.pro/classes/class_subsystem_text.html?highlight=text_finished#signal-text_finished), [about_to_show_text](https://docs.dialogic.pro/classes/class_subsystem_text.html?highlight=text_finished#signal-about_to_show_text), and [continued_revealing_text](https://docs.dialogic.pro/classes/class_dialogicnode_dialogtext.html?highlight=continued_revealing_text#signal-continued_revealing_text).

My portraits start talking when `about_to_show` fires and finishes talking when `text_finished` is emitted.
I use a mouth layer that occurs above my base sprite layer, the mouth sprite is an animation sheet and I use Godot's `Sprite2D` for this. 

Last but not least, you can use the `_unhighlight` and `_highlight` base methods on the `DialogicPortrait` to identify which character is speaking and prevent others from animating their mouth.

Let me know if this helps!

**Reply by Jowan-Spooner** [reply]
You can also just compare the `character` property on the portrait class to the `'character'` entry in the info dictionary given by about_to_show_text to know if the character is actually talking (without _unhighlight and _highlight).

**Reply by TunaNoot** [reply]
I may need to look more into the documentation. I've made a scene with a Node2D as the root node with a Sprite2D as its child for the body's "base", and two AnimatedSprite2D's - the eyes and mouth as children of the body sprite. The mouth has two animations, one for when the character is silent and one for when the character is speaking. Then, I've set this scene as my character's portrait.

I've then added a script that only has one line so far, stating that it extends DialogicPortrait. I've made a test timeline with this character and the sprites aren't displaying. I'm pretty sure I'm missing something.

**Reply by Jowan-Spooner** [reply]
Yes. You need to overwrite at least the method _get_covered_rect() so dialogic knows how big your scene is and can place it in the PortraitContainer.
